src/partisan/irods.py

```diff
@@ -822,18 +822,41 @@ class AVU(object):
     """The attribute history suffix"""
 
     def __init__(self, attribute: Any, value: Any, units=None, namespace=None):
-        if namespace:
-            if namespace.find(AVU.SEPARATOR) >= 0:
-                raise ValueError(
-                    f"AVU namespace '{namespace}' contained '{AVU.SEPARATOR}'"
-                )
+        """Create a new AVU instance.
+
+        Args:
+            attribute: The attribute to use. If this is not a string, the string
+               representation of this argument becomes the attribute.
+            value: The value to use. If this is not a string, the string
+               representation of this argument becomes the value.
+            units: The units to use. Optional, defaults to None.
+            namespace: The namespace (prefix) to be used for the attribute. Optional,
+               but if supplied, must be the same as any existing namespace on the
+               attribute string.
+        """
         if attribute is None:
             raise ValueError("AVU attribute may not be None")
         if value is None:
             raise ValueError("AVU value may not be None")
+        if namespace is None:
+            namespace = ""
+
+        attr = str(attribute)
+        if attr.find(AVU.SEPARATOR) >= 0:
+            ns, at = attr.split(AVU.SEPARATOR, maxsplit=1)
+            if namespace and ns != namespace:
+                raise ValueError(
+                    f"AVU attribute namespace '{ns}' did not match "
+                    f"the declared namespace '{namespace}' for "
+                    f"attribute '{attr}', value '{value}'"
+                )
+            namespace, attr = ns, at
+
+        if namespace.find(AVU.SEPARATOR) >= 0:
```

Comment:
Agreed, that's not right. That test was there before this change and is meant to test that a namespace supplied as a keyword arg doesn't have a separator in it. I should move it earlier so that the namespace being checked is clearly the one passed in, not one that's been created from the attribute because a namespace wan't passed in.